src/components/calcite-checkbox/calcite-checkbox.tsx

```diff
@@ -197,12 +197,19 @@ export class CalciteCheckbox {
   }
 
   render() {
+    const hasLabel = this.el.textContent ? true : false;
     return (
-      <Host role="checkbox" aria-checked={this.checked.toString()}>
+      <Host role="checkbox" aria-checked={this.checked.toString()} class={{ hasLabel }}>
```

Comment:
Is there a class you can add in shadow dom instead of the host? It's usually a best practice to not add classes on the host element because they are exposed to the user and can be modified/removed. It may warrant a div container within the host.

Reply:
I guess I can make a child `div` directly under the host element to contain everything, but that felt unnecessary to add, but maybe it makes sense to do.

The other option is I can convert the class to an attribute and use an attribute selector in the CSS to get the job done.  Which would you prefer?

Reply:
I think I prefer adding a container within the host but what does everyone else think?

Reply:
I'm doing this same thing in the radio button component, so if everyone agrees on using a container div instead of an attribute, I'll update checkbox and radio button to match.

Reply:
> I think I prefer adding a container within the host but what does everyone else think?

👍 for the container-approach since it helps w/ encapsulation.

Reply:
I know an extra div sucks, but I think it's worth it to keep things encapsulated instead of being exposed to the user.

Reply:
Yep, I agree!  It works out better this way too because I can forego rendering the container div when there's no label, so that's actually a win-win.

Reply:
I'll push this up soon so you can re-check.